src/migrations/FolderMigration.ts

```diff
@@ -1,25 +1,37 @@
-import PathResolver from '../r2mm/manager/PathResolver';
 import * as path from 'path';
 import FsProvider from '../providers/generic/file/FsProvider';
-import FileUtils from '../utils/FileUtils';
+import PathResolver from '../r2mm/manager/PathResolver';
 import CacheUtil from '../r2mm/mods/CacheUtil';
+import FileUtils from '../utils/FileUtils';
 
-export default class FolderMigration {
+/**
+ * Mod directory structure was changed when support for other games
+ * besides RoR2 was added. Update the dir structure if the old one is
+ * still in use.
+ */
+export class FolderMigration {
 
-    public static async needsMigration(): Promise<boolean> {
+    public static async needsMigration() {
         const fs = FsProvider.instance;
         return await fs.exists(path.join(PathResolver.ROOT, "mods"));
     }
 
-    public static async runMigration(): Promise<void> {
-        console.log("Started migration");
+    public static async runMigration() {
+        if (!await this.needsMigration()) {
+            return;
+        }
+
+        console.log("Started legacy directory migration");
         const fs = FsProvider.instance;
+        const rorPath = path.join(PathResolver.ROOT, "RiskOfRain2");
         await CacheUtil.clean();
```

Comment:
I was wondering why is cache clean run as a part of the migration? Not that it matters that much, I guess.

Reply:
Originally it copy/pasted the directory, however there were reports that it was failing silently in some cases causing loss of profiles.

It got reworked to rename instead and that was just leftover.

Doesn't hurt to leave in, however no preference if you want to remove it or not

Reply:
I guess I'll leave it in just in case, to avoid any regressions.

Reply:
So I ended up removing it anyway, because I noticed that it caused unhandled exceptions. [Reading the profile folder](https://github.com/ebkr/r2modmanPlus/blob/develop/src/r2mm/mods/CacheUtil.ts#L20) fails if the migrations are not run yet, because it makes wrong assumptions about the directory structure. I'm _quite_ sure that this is not caused by my changes, but was a pre-existing condition.